packages/devtools_app/lib/src/debugger/file_search.dart

```diff
@@ -42,40 +43,58 @@ class _FileSearchFieldState extends State<FileSearchField>
   void initState() {
     super.initState();
 
-    widget.autoCompleteController.currentDefaultIndex = 0;
+    autoCompleteController = AutoCompleteController()..currentDefaultIndex = 0;
 
     addAutoDisposeListener(
-        widget.autoCompleteController.searchNotifier, _handleSearch);
-    addAutoDisposeListener(
-        widget.autoCompleteController.searchAutoCompleteNotifier,
+        autoCompleteController.searchNotifier, _handleSearch);
+    addAutoDisposeListener(autoCompleteController.searchAutoCompleteNotifier,
         _handleAutoCompleteOverlay);
 
-    _query = widget.autoCompleteController.search;
+    _query = autoCompleteController.search;
     _matches = widget.debuggerController.sortedScripts.value;
 
     // Open the autocomplete results immediately before a query is entered:
     SchedulerBinding.instance.addPostFrameCallback((_) => _handleSearch());
   }
 
+  @override
+  Widget build(BuildContext context) {
+    return buildAutoCompleteSearchField(
```

Comment:
Hmm I'm not sure I understand. This is a pre-existing method in `search.dart`: https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/ui/search.dart#L533, I don't think it can be easily converted

Reply:
What Jacob is suggesting is to basically take the buildAutoCompleteSearchField method and convert it to 

`class AutoCompleteSearchField extends StatefulWidget {` where the parameters in the constructor match the parameters for this method. Then the build method does the work of constructing this widget tree..

However, in this case, I think we should actually leave this as is, because we only want classes that mixin `SearchFieldMixin` to be using this auto complete search field widget, and we can't enforce that if this is a public widget class. I suppose we could make a private widget _AutoCompleteSearchField and have this `buildAutoCompleteSearchField` method return that private widget. That would be a good solution.

Reply:
Ah got it! I've left a TODO for having it return a private ` _AutoCompleteSearchField`. Will address that in a separate PR.